github.js

```diff
@@ -18,23 +18,26 @@ const GITHUB_AUTH_HEADERS = {
 /**
  * @param {String} filePath - path where config file is stored
  * @param {String} repoName - name of the github repository
+ * @param {boolean} isCustomPath - true if users pass a path flag to the CLI
+ * @param {String | undefined} - branch to fetch from. if undefined, default branch used
  * @returns {Buffer|Null} Zip data buffer
  */
 async function fetchJsonFromRepository(
-  repoName = 'HubSpot/hubspot-project-components',
+  repoName,
   filePath,
-  repoPath = false
+  ref,
+  isCustomPath = false
 ) {
   try {
-    const URI = `https://raw.githubusercontent.com/${repoName}/${filePath}`;
+    const URI = `https://raw.githubusercontent.com/${repoName}/${ref}/${filePath}`;
     logger.debug(`Fetching ${URI}...`);
 
     return await request.get(URI, {
       json: true,
       headers: { ...DEFAULT_USER_AGENT_HEADERS, ...GITHUB_AUTH_HEADERS },
     });
   } catch (err) {
-    if (repoPath && err.statusCode === 404) {
+    if (isCustomPath && err.statusCode === 404) {
```

Comment:
If we're refactoring this anyways, maybe this function shouldn't be the thing that handles exiting the process and logging. It might simplify the args that need to be passed in if we force the CLI to handle the request error (i.e. we may not need `isCustomPath` anymore)

I wrote ^ but then realized doing that would require us to catch the errors in the CLI in every place that we call `fetchJsonFromRepository`, and maybe that's annoying enough that it makes sense to continue to catch the errors here?

Reply:
Yeah that's a good point. That's actually how its being done in `local-dev-lib`, so I was thinking we could just keep this for now and the change will happen when we switch to `local-dev-lib`